src/backend/distributed/commands/vacuum.c

```diff
@@ -73,29 +78,97 @@ static CitusVacuumParams VacuumStmtParams(VacuumStmt *vacstmt);
 void
 PostprocessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
 {
-	int relationIndex = 0;
-	List *vacuumRelationList = ExtractVacuumTargetRels(vacuumStmt);
-	List *relationIdList = NIL;
 	CitusVacuumParams vacuumParams = VacuumStmtParams(vacuumStmt);
+	const char *stmtName = (vacuumParams.options & VACOPT_VACUUM) ? "VACUUM" : "ANALYZE";
+
+	/*
+	 * No table in the vacuum statement means vacuuming all relations
+	 * which is not supported by citus.
+	 */
+	if (list_length(vacuumStmt->rels) == 0)
+	{
+		/* WARN for unqualified VACUUM commands */
+		ereport(WARNING, (errmsg("not propagating %s command to worker nodes", stmtName),
+						  errhint("Provide a specific table in order to %s "
+								  "distributed tables.", stmtName)));
+	}
+
+
+	List *citusRelationIdList = VacuumCitusRelationIdList(vacuumStmt, vacuumParams);
+	if (list_length(citusRelationIdList) == 0)
+	{
+		return;
+	}
```

Comment:
Sorry, it's a bit late review but shouldn't this come before below ?
https://github.com/citusdata/citus/pull/4221/files#diff-35b5b7bc55390cdb63bb4783fd162d59R88-R94

Reply:
Umm, practically yeah. I guess it is the question of:

"Do we want `VACUUM;` to error always or only if there is at least one citus table" 

I guess it could be better not to error when there is no citus table.

Reply:
I agree for that
>I guess it could be better not to error when there is no citus table.

Reply:
This (https://github.com/citusdata/citus/pull/4221#discussion_r502406920) really doesn't make sense, never mind